feat: [Major version bump] Merge breaking changes into main (#363)
DataDog/datadog-cdk-constructs: README.md

The Datadog CDK Construct Libraries use Projen to maintain project configuration files such as the `package.json`, `.gitignore`, `.npmignore`, etc. Most of the configuration files will be protected by Projen via read-only permissions. In order to change these files, edit the `.projenrc.js` file, then run `npx projen` to synthesize the new changes. Check out [Projen][13] for more details.

## Migrating from v2-1.x.x to v2-2.x.x

> We bumped the major version from `1.x.x` to `2.x.x` in February 2025. Below are the changes needed to migrate to the new version.

Reviewer: ```suggestion
In February 2025, Datadog released a major version update from `1.x.x` to `2.x.x`. The required changes to migrate to the new version are as follows:
```

1. Rename the classes for instrumenting Lambda functions:

   1. `Datadog` -> `DatadogLambda`
   2. `DatadogProps` -> `DatadogLambdaProps`

>       For examples, see [Usage](#usage) section and [examples/](https://github.com/DataDog/datadog-cdk-constructs/tree/main/examples) folder.

Reviewer: ```suggestion
      For examples, see the [Usage](#usage) section and [examples/](https://github.com/DataDog/datadog-cdk-constructs/tree/main/examples) folder.
```
Could you please link to https://github.com/DataDog/datadog-cdk-constructs/tree/main/examples using a reference style link? It seems like we are inconsistent with that on this page in general, but it's important for the docs translation service.

> 2. Upgrade Node.js version to 18.18.0 or above.

Reviewer: ```suggestion
2. Upgrade Node.js version to `18.18.0` or above.
```
